test/pop/blockchain/vbkblock_finalization_test.cpp

```diff
@@ -72,10 +71,17 @@ TEST_F(VbkBlockFinalization, OverBtcLimitTest) {
 
   ASSERT_EQ(btctip->getHash(), tree->btc().getBestChain().tip()->getHash());
 
+  // save state
+  auto batch = popminer->getStorage().generateWriteBatch();
+  auto writer = adaptors::BlockBatchImpl(*batch);
+  saveTree(*tree, writer);
+  saveTree(tree->btc(), writer);
+  batch->writeBatch();
```

Comment:
What if state was not saved? Will it die?

Reply:
generally yes, so for the correct work with the finalization you should to save the state